compiler/codegen/codegen.cpp

```diff
@@ -2183,15 +2183,15 @@ static const char* generateFileName(ChainHashMap<char*, StringHashFns, int>& fil
 
   // create the lowercase filename
   char lowerFilename[FILENAME_MAX];
-  sprintf(lowerFilename, "%s", currentModuleName);
+  snprintf(lowerFilename, sizeof(lowerFilename), "%s", currentModuleName);
```

Comment:
I know my previous comment said to use `sizeof`, but if you are preferring to use the variable when it's right there, you could use FILENAME_MAX here. I stand by `sizeof` though.

Reply:
In this case I preferred `sizeof` because `FILENAME_MAX` is a constant that isn't specific to this buffer, and it would make perfect sense if we wanted to do `FILENAME_MAX-1` or something like that.